docs/guide/retryable.md

```diff
@@ -6,120 +6,26 @@ poorly written code which should be fixed first. But they can also fail due to e
-jobs, see the following method.
-
-RetryableJobInterface
+AttemptsRestrictedPayloadInterface
 ---------------------
 
-To have more control over the retry logic a job can implement the `RetryableJobInterface`.
-For example:
-
 ```php
-class SomeJob extends BaseObject implements RetryableJobInterface
+class RetryablePayload extends SimplePayload implements AttemptsRestrictedPayloadInterface
 {
-    public function execute($queue)
-    {
-        //...
-    }
-
-    public function getTtr()
+    public function getAttempts(): int
     {
-        return 15 * 60;
+        return 2;
     }
 
-    public function canRetry($attempt, $error)
+    public function getName(): string
     {
-        return ($attempt < 5) && ($error instanceof TemporaryException);
+        return 'retryable';
     }
 }
 ```
 
-The `getTtr()` and `canRetry()` methods have a higher priority than the component options mentioned above.
-
-Event handlers
---------------
-
-The third method to control TTR and number of retries for failed jobs involves the
-`Queue::EVENT_BEFORE_PUSH` and `Queue::EVENT_AFTER_ERROR` events.
-
-The TTR can be set with the `Queue::EVENT_BEFORE_PUSH` event:
-
-```php
-Yii::$app->queue->on(Queue::EVENT_BEFORE_PUSH, function (PushEvent $event) {
-    if ($event->job instanceof SomeJob) {
-        $event->ttr = 300;
-    }
-});
-```
-
-And the `Queue::EVENT_AFTER_ERROR` event can be used to decide whether to try another attempt:
-
-```php
-Yii::$app->queue->on(Queue::EVENT_AFTER_ERROR, function (ExecEvent $event) {
-    if ($event->job instanceof SomeJob) {
-        $event->retry = ($event->attempt < 5) && ($event->error instanceof TemporaryException);
-    }
-});
-```
-
-Event handlers are executed after the `RetryableJobInterface` methods, and therefore have the highest
-priority.
-
 Restrictions
 ------------
 
-Full support for retryable jobs is implemented in the [Beanstalk], [DB], [File], [AMQP Interop] and [Redis] drivers.
-The [Sync] driver will not retry failed jobs. The [Gearman] driver doesn't support retryable jobs.
-[RabbitMQ] has only its basic retryable support, where the number of attempts can not be set.
-
-[AWS SQS] uses [Dead Letter Queue] for handling messages that were failed to process.
-All unprocessed messages after a maximum number of attempts are moved to that queue.
-You should set an address of a Dead Letter Queue and a maximum number of attempts in the AWS Console while creating a queue.
-
-[Beanstalk]: driver-beanstalk.md
-[DB]: driver-db.md
-[File]: driver-file.md
-[Redis]: driver-redis.md
-[Sync]: driver-sync.md
-[Gearman]: driver-gearman.md
-[RabbitMQ]: driver-amqp.md
-[AMQP Interop]: driver-amqp-interop.md
-[AWS SQS]: driver-sqs.md
-[Dead Letter Queue]: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.html
+Full support for retryable payload is implemented in the [AMQP Interop](https://github.com/yiisoft/yii-queue-amqp) driver.
```

Comment:
```suggestion
Full support for retryable payload is implemented in the [AMQP](https://github.com/yiisoft/yii-queue-amqp) driver.
```
AMQP Interop driver is not implemented because [queue-interop package](https://github.com/queue-interop/queue-interop) is not maintained anymore.